src/java.base/share/classes/sun/security/pkcs/PKCS9Attribute.java

```diff
@@ -179,178 +179,126 @@ public class PKCS9Attribute implements DerEncoder {
-                Class.forName("java.lang.String");
-            VALUE_CLASSES[8] = str;   // UnstructuredAddress
-            VALUE_CLASSES[9] = null;  // ExtendedCertificateAttributes
-            VALUE_CLASSES[10] = null;  // IssuerAndSerialNumber
-            VALUE_CLASSES[11] = null;  // not used
-            VALUE_CLASSES[12] = null;  // not used
-            VALUE_CLASSES[13] = null;  // not used
-            VALUE_CLASSES[14] =        // ExtensionRequest
-                Class.forName("sun.security.x509.CertificateExtensions");
-            VALUE_CLASSES[15] = null;  // not supported yet
-            VALUE_CLASSES[16] = null;  // not supported yet
-            VALUE_CLASSES[17] = BYTE_ARRAY_CLASS;  // SignatureTimestampToken
-            VALUE_CLASSES[18] = BYTE_ARRAY_CLASS;  // CMSAlgorithmProtection
+            add(EMAIL_ADDRESS_OID, false, str, DerValue.tag_IA5String);
+
+            add(UNSTRUCTURED_NAME_OID, false, str,
+                DerValue.tag_IA5String,
+                DerValue.tag_PrintableString,
+                DerValue.tag_T61String,
+                DerValue.tag_BMPString,
+                DerValue.tag_UniversalString,
+                DerValue.tag_UTF8String);
+
+            add(CONTENT_TYPE_OID, true,
+                Class.forName("sun.security.util.ObjectIdentifier"),
+                DerValue.tag_ObjectId);
+
+            add(MESSAGE_DIGEST_OID, true, byte[].class,
+                DerValue.tag_OctetString);
+
+            add(SIGNING_TIME_OID, true, Class.forName("java.util.Date"),
+                DerValue.tag_UtcTime, DerValue.tag_GeneralizedTime);
+
+            add(COUNTERSIGNATURE_OID, false,
+                Class.forName("[Lsun.security.pkcs.SignerInfo;"),
+                DerValue.tag_Sequence);
+
+            add(CHALLENGE_PASSWORD_OID, true,
+                Class.forName("java.lang.String"),
+                    DerValue.tag_PrintableString,
+                    DerValue.tag_T61String,
+                    DerValue.tag_BMPString,
+                    DerValue.tag_UniversalString,
+                    DerValue.tag_UTF8String);
+
+            add(UNSTRUCTURED_ADDRESS_OID, false, str,
+                DerValue.tag_PrintableString,
+                DerValue.tag_T61String,
+                DerValue.tag_BMPString,
+                DerValue.tag_UniversalString,
+                DerValue.tag_UTF8String);
+
+            add(EXTENSION_REQUEST_OID, true,
+                Class.forName("sun.security.x509.CertificateExtensions"),
+                DerValue.tag_Sequence);
+
+            add(SIGNING_CERTIFICATE_OID, true,
+                Class.forName("sun.security.pkcs.SigningCertificateInfo"),
+                DerValue.tag_Sequence);
+
+            add(SIGNATURE_TIMESTAMP_TOKEN_OID, true, byte[].class,
+                DerValue.tag_Sequence);
+
+            add(CMS_ALGORITHM_PROTECTION_OID, true, byte[].class,
+                DerValue.tag_Sequence);
+
         } catch (ClassNotFoundException e) {
```

Comment:
Probably can remove this try/catch block if you make above changes to not call `Class.forName`.